api/views.py

```diff
@@ -815,3 +820,103 @@ def create(self, request, *args, **kwargs):
         except Exception as err:
             print(err)
         return HttpResponse("Created new template: {}.".format(name), status=200)
+
+
+@api_view(["GET", "POST"])
+@permission_classes(
+    (
+        IsAuthenticated,
+        AdminPermission,
+    )
+)
+def update_app_status(request):
+    """
+    Manages the app instance status.
+    Implemented as a DRF function based view.
+    Supports GET and POST verbs.
+
+    The service contract for the POST actions is as follows:
+    :param release str: The release id of the app instance.
+    :param new-status str: The new status code.
```

Comment:
Great question. This is however a free text field and not pre-defined as we rely on the codes coming from k8s.

Reply:
Exactly, we want to capture ALL statuses coming from K8s. There are no list (that i can find) of all possible statuses. If you find one, i will update the status map in the event-listener. Right now we map this as follow:
```
K8S_STATUS_MAP = {
    "CrashLoopBackOff": "Error",
    "Completed": "Retrying...",
    "ContainerCreating": "Created",
    "PodInitializing": "Pending",
    "ErrImagePull": "Image Error",
    "ImagePullBackOff": "Image Error",
}
```
But it would be nice with a complete status map.